rust/processor/src/db/common/models/emojicoin_models/models/market_registration_event.rs

```diff
@@ -0,0 +1,126 @@
+use crate::db::common::models::emojicoin_models::json_types::{
+    MarketRegistrationEvent, StateEvent, TxnInfo,
+};
+use crate::db::common::models::emojicoin_models::{enums, utils::micros_to_naive_datetime};
+use crate::schema::market_registration_events;
+use field_count::FieldCount;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize)]
+#[diesel(primary_key(market_id))]
+#[diesel(table_name = market_registration_events)]
+pub struct MarketRegistrationEventModel {
+    // Transaction metadata.
+    pub transaction_version: i64,
+    pub sender: String,
+    pub entry_function: Option<String>,
+    pub transaction_timestamp: chrono::NaiveDateTime,
+
+    // Market and state metadata.
+    pub market_id: i64,
+    pub symbol_bytes: Vec<u8>,
+    pub bump_time: chrono::NaiveDateTime,
+    pub market_nonce: i64,
+    pub trigger: enums::Trigger,
+
+    // Market registration event data.
+    pub registrant: String,
+    pub integrator: String,
+    pub integrator_fee: i64,
+}
+
+impl MarketRegistrationEventModel {
+    pub fn new(
+        txn_info: TxnInfo,
+        market_registration_event: MarketRegistrationEvent,
+        state_event: StateEvent,
+    ) -> MarketRegistrationEventModel {
+        let MarketRegistrationEvent {
+            market_metadata,
+            time,
+            registrant,
+            integrator,
+            integrator_fee,
+            ..
+        } = market_registration_event;
+
+        MarketRegistrationEventModel {
+            // Transaction metadata.
+            transaction_version: txn_info.version,
+            sender: txn_info.sender.clone(),
+            entry_function: txn_info.entry_function.clone(),
+            transaction_timestamp: txn_info.timestamp,
+
+            // Market and state metadata.
+            market_id: market_metadata.market_id,
+            symbol_bytes: market_metadata.emoji_bytes,
+            bump_time: micros_to_naive_datetime(time),
+            market_nonce: state_event.state_metadata.market_nonce,
+            trigger: state_event.state_metadata.trigger,
+
+            // Market registration event data.
+            registrant,
+            integrator,
+            integrator_fee,
+        }
+    }
+}
+
+// NOTE: We don't store most of the state event data for a market registration event
+// because it is largely identical for every single market, and we never query by that data.
+// The only fields that are variable are:
+//   - emoji_bytes
+//   - market_address
+//   - market_id
+//   - bump_time
+//
+// If this ever changes, we can easily update the rows retroactively with a join on the market_id
+// with a market nonce of 1.
+
+// The StateEvent data emitted alongside a MarketRegistrationEvent looks like this:
+/*
```

Comment:
Is this still needed?

Commented out code is a smell

Reply:
It's more json data than code, but I also clarified the intent behind leaving it there. I left it there for now to document why I made the table model the way I did. 

We can always move the JSON data to a unit test and point to it there instead of having it here